autoload/test/viml/themis.vim

```diff
@@ -0,0 +1,38 @@
+if !exists('g:test#viml#themis#file_pattern')
+  let g:test#viml#themis#file_pattern = '.vim$'
+endif
+
+if !exists('g:test#viml#themis#executable')
+  let g:test#viml#themis#executable = 'themis'
+endif
+
+function! test#viml#themis#test_file(file) abort
+  return a:file =~# g:test#viml#themis#file_pattern
+        \ && !empty(filter(readfile(a:file), 'v:val =~# ''\<themis#suite\s*('''))
+endfunction
+
+function! test#viml#themis#build_position(type, position) abort
+  if a:type == 'nearest' || a:type == 'file'
+    return [a:position['file']]
+  else
+    return []
+  endif
+endfunction
+
+function! test#viml#themis#build_args(args) abort
+  let args = a:args
+
+  if exists('test#viml#themis#args')
+    let args = g:test#viml#themis#args + args
+  endif
+
+  return args
+endfunction
+
+function! test#viml#themis#executable() abort
+  if !executable(g:test#viml#themis#executable)
```

Comment:
I think this might throw an exception if `g:test#viml#themis#executable` is undefined. You can actually remove all of the code except the `throw`, because when `g:test#{language}#{runner}#executable` is defined, then the corresponding `g:test#{language}#{runner}#executable()` function is not called.

Since we won't be actually testing whether what's assigned is executable, we could rephrase the error message to be more straightforward:

```vim
function! test#viml#themis#executable() abort
  throw 'The `g:test#viml#themis#executable` variable needs to be set with the path to the "themis" executable'
endfunction
```

Reply:
I made this change but the tests are hitting the function and throwing. It seems like maybe it's calling the function but throwing away the return value if the executable variable is set? Changing this to `return 'themis'` seems to work as expected.

Reply:
Oops, the function was still being executed (even though it's not used), I pushed a change now to `master` which doesn't call the function if the variable is defined.